File: mixin-sleman-webface/src/main/js/sleman/model/XForeach.ts
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import EReference from "webface/model/EReference";

import SForeach from "sleman/SForeach";

import Printer from "sleman/model/Printer";
import * as model from "sleman/model/model";
import XEvaluation from "sleman/model/XEvaluation";
import XExpression from "sleman/model/XExpression";

export default class XForeach extends XEvaluation implements SForeach {

    public static XCLASSNAME: string = model.getEClassName("XForeach");

    public static FEATURE_EXPRESSION = new EReference("expression", XExpression);

    private expression: XExpression = null;

    constructor() {
        super(model.createEClass(XForeach.XCLASSNAME), [
            XForeach.FEATURE_EXPRESSION,
        ]);
    }

    public getExpression(): XExpression {
        return this.expression;
    }

    public setExpression(newExpression: XExpression): void {
        let oldExpression = this.expression;
        this.expression = newExpression;
        this.eSetNotify(XForeach.FEATURE_EXPRESSION, oldExpression, newExpression);
    }

    public print(printer: Printer): void {
        printer.term("foreach");
        printer.space();
        printer.part(this.expression);
    }

    public toString(): string {
        return this.toLiteral();
    }

}

File: mixin-sleman/src/main/java/com/andia/mixin/sleman/model/XForeach.java
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>. */
package com.andia.mixin.sleman.model;

import com.andia.mixin.model.EFeature;
import com.andia.mixin.model.EReference;
import com.andia.mixin.sleman.api.SForeach;

public class XForeach extends XStructure implements SForeach {

	public static String XCLASSNAME = Sleman.getEClassName("XForeach");

	public static EReference FEATURE_EXPRESSION = new EReference("expression", XExpression.class);

	private XExpression expression = null;

	public XForeach() {
		super(Sleman.createEClass(XForeach.XCLASSNAME), new EFeature[] {
				XForeach.FEATURE_EXPRESSION,
		});
	}

	@Override
	public XExpression getExpression() {
		return this.expression;
	}

	public void setExpression(XExpression newExpression) {
		XExpression oldExpression = this.expression;
		this.expression = newExpression;
		this.eSetNotify(FEATURE_EXPRESSION, oldExpression, newExpression);
	}

	@Override
	public String toLiteral() {
		return "foreach " + this.expression.toLiteral();
	}

	@Override
	public String toString() {
		return this.toLiteral();
	}

}
